File: java-service/src/main/kotlin/com/ciandt/hello/core/usecase/sayhello/SayHelloUseCase.kt
package com.ciandt.hello.core.usecase

import com.ciandt.hello.core.entity.HelloMessage
import org.springframework.beans.factory.annotation.Autowired

class SayHelloUseCase {

    var gw: GetRealNameToHello

    @Autowired
    constructor(gw: GetRealNameToHello) {
        this.gw = gw
    }

    fun sayHello(name: String): HelloMessage {
        val realName = gw.getRealNameToHello(name = name)
        return HelloMessage(message = "Hello $realName!")
    }
}

File: python-service/core/usecase/say_hello_usecase.py
from core.domain.hello_message import HelloMessage


class SayHelloUseCase:

    def __init__(self, real_name_gw):
        self._real_name_gw = real_name_gw

    def say_hello(self, name):
        return HelloMessage(msg=self._real_name_gw.get_real_name(name))
